types/ramda/index.d.ts

```diff
@@ -1000,6 +1000,7 @@ declare namespace R {
          * additional call to fn; instead, the cached result for that set of arguments will be returned.
          */
         memoize(fn: (...a: any[]) => any): (...a: any[]) => any;
+        memoize<T>(fn: (...a: any[]) => T): (...a: any[]) => T;
```

Comment:
The order of memoize function overload needs to be reversed so that the default signature is picked up ?

Reply:
This also could be written as the following, without overloads:

```ts
  memoize<T = any>(fn: (...a: any[]) => T): (...a: any[]) => T;
```